### PR TITLE
De-groovy'ed fiat-roles, since  java 8 and groovy don't play nicely.

### PR DESCRIPTION
**Pull request by ttomsu**
First off: :tada: :fireworks: 

Groovy -> java is a huge PITA, so it's possible I may have missed something in the UserRoles providers. 

@jtk54 @gbougeard @duftler PTAL

@ajordens @cfieber FYI

### CODE REVIEW

File: fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+import java.util.*;
```
Comment (ajordens) [inline fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java]: Think we still ignore the .*'s in Java

File: fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    Boolean isMemberOfOrg = false;
+
+    try {
+      Response response = master.getGitHubClient()
+                                .isMemberOfOrganization(gitHubProperties.getOrganization(),
+                                                        userName);
+      isMemberOfOrg = (response.getStatus() == 204);
+    } catch (RetrofitError e) {
+      if (e.getKind() == RetrofitError.Kind.NETWORK) {
+        log.error("Could not find the server {master.baseUrl}", e);
```
Comment (ajordens) [inline fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java]: There's no slf4j magic that makes this template work is there?

Similarly below I see some `${master.baseUrl}`

**Comment by ajordens** [comment]
I could (possibly?) be convinced that groovy `fiat-google` and `fiat-github` artifacts aren't the end of the world. 

Not sure if we gain a tremendous amount from groovy'ing -> java'ing a bunch of code, particularly other OSS contributions.

It's possibly I'm the only one that thinks this and that the effort is worthwhile.

File: fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/client/GitHubClient.java [hunk 3]
```diff
@@ -14,33 +14,35 @@
...
    */
-  @GET('/user/teams')
-  List<GitHubMaster.Team> getUserTeams()
+  @GET("/user/teams")
+  List<GitHubMaster.Team> getUserTeams();
 
-  @GET('/orgs/{org}/teams')
-  List<GitHubMaster.Team> getOrgTeams(@Path('org') String org)
+  @GET("/orgs/{org}/teams")
+  List<GitHubMaster.Team> getOrgTeams(@Path("org") String org);
```
Comment (gbougeard) [inline fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/client/GitHubClient.java]: fyi there were some modifications on this on gate, I don't know how/when you want to proceed to apply them in fiat.

**Comment by duftler** [comment]
Can you describe the motivation for this change? What troubles did you run into between groovy/java 8?

**Comment by ttomsu** [comment]
@duftler Specifically, the groovy compiler barfs on Java 8 files with closures, specifically the `->` arrow operator. I didn't (but should) have tested server startup in https://github.com/spinnaker/fiat/pull/55. 

@ajordens I wouldn't be opposed to it, but I need to use the time between now an EOQ to work on integrations. We can split `fiat-roles` at any time.

File: fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/client/GitHubClient.java (hunk 3 above)
Comment (ttomsu) [inline fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/client/GitHubClient.java]: Filed https://github.com/spinnaker/fiat/issues/59

File: fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java (hunk 2 above)
Comment (ttomsu) [inline fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java]: Good catch. I meant to go back and fix those. Done.

File: fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java (hunk 1 above)
Comment (ttomsu) [inline fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java]: Changed my IDE to always expand .* references.

**Comment by ajordens** [comment]
@ttomsu Gotcha ... my suggestion was trying to make your life easier, specifically around not groovy -> java8'ing a bunch of non-core code unless there was a strong reason to feel the pain.

**Comment by duftler** [comment]
LGTM.

**Pull request merged**